session.h

```diff
@@ -50,6 +50,24 @@ typedef struct {
   } addr;                   /**< session IP address and port */
   int ifindex;              /**< network interface index */
 } session_t;
+#elif defined(WITH_LWIP)
+
+#include "lwip/sockets.h"
+#undef write
+#undef read
+typedef struct {
+  socklen_t size;               /**< size of addr */
+  union {
+    struct sockaddr     sa;
+    struct sockaddr_storage st;
+    struct sockaddr_in  sin;
+    struct sockaddr_in6 sin6;
+  } addr;
+  int ifindex;
+} session_t;
+
+typedef unsigned char uint8_t;
+
 #else /* ! WITH_CONTIKI && ! WITH_RIOT_SOCK */
```

Comment:
Maybe " && ! WITH_LWIP"?

Reply:
Refactored order of code to reduce code duplication.